### PR TITLE
Context templating in external links

### LINKED ISSUE #5290: Context templating ({context.*} and {i18n.*}) not resolved in external link URLs in core-modular
## Description

In the old Luigi Core, external link nodes with URLs containing `{context.someValue}` and `{i18n.currentLocale}` were properly resolved before rendering as `<a href="...">` in the side navigation.

In core-modular, `openExternalLink()` in `navigation-helpers` only resolves `:pathParam`-style variables via `replaceVars(n, t, ":", false)`. The `{context.*}` and `{i18n.*}` placeholders are never substituted.

## Reproduction

Config:
```js
{
  label: 'Context Value Replacement',
  context: { someValue: 'bar' },
  externalLink: {
    url: 'http://sap.com/{i18n.currentLocale}?foo={context.someValue}',
    sameWindow: true
  }
}
```

**Expected:** Clicking the node navigates to `http://sap.com/en?foo=bar`
**Actual:** Navigates to `http://sap.com/{i18n.currentLocale}?foo={context.someValue}` (unresolved)

### PR DESCRIPTION
**Pull request by walmazacn**
**Description**

Changes proposed in this pull request:

- fixes issue with context templating in external links

**Related issue(s)**

Fixes #5290

### CODE REVIEW

**Review by JohannesDoberer — CHANGES_REQUESTED** [review]
There are two issues and poth related to each other.

### Context not propagated to external link node — `{context.*}` placeholders remain unresolved
#### Problem

`substituteViewUrl` resolves `{context.*}` placeholders by looking at `node.context`:

```ts
if (node.context) {
  viewUrl = GenericHelpers.replaceVars(viewUrl, node.context, contextVarPrefix);
}
```

However, the external link node **does not inherit** the resolved context from its parent. In `getPathData`, context is only set on nodes that are **part of the navigation path** (i.e. traversed by the loop). The external link node is a sibling/child that the user clicks from the left nav — it never passes through `getPathData`'s loop, so `node.context` is either `undefined` or whatever was declared statically in the config.

This means `{context.tenant}` is never substituted and gets URL-encoded to `%7Bcontext.tenant%7D`.

#### Steps to reproduce

**Config:**
```js
{
  pathSegment: ':tenantId',
  context: { tenant: ':tenantId' },
  children: [
    {
      label: 'External Docs',
      pathSegment: 'docs',
      externalLink: {
        url: 'https://example.com/{context.tenant}/docs'
      }
    }
  ]
}
```

1. Navigate to `/home/tenantA/overview`
2. Click "External Docs"
3. **Expected:** `https://example.com/tenantA/docs`
4. **Actual:** `https://example.com/%7Bcontext.tenant%7D/docs`

#### Root cause

`navItemClick` receives `pathData` (which contains the fully resolved, accumulated context at `pathData.context`), but spreads only the node itself into `nodeData`. Since the external link node has no own resolved context, `substituteViewUrl` finds nothing to replace.


### Node mutation in `navItemClick` causes stale external link URLs
#### Problem

This line mutates the cached node object directly. Since nodes are stored by reference in `NodeDataManagementService`, the original URL template (e.g. `https://example.com/{context.tenant}/docs`) is permanently overwritten with the resolved value after the first click.

On subsequent clicks — even with a different context or pathParams — the template is gone, so `substituteViewUrl` has nothing left to resolve.

#### Steps to reproduce

No context-switcher needed — a dynamic pathParam is sufficient:

**Config:**
```js
{
  pathSegment: ':tenantId',
  context: { tenant: ':tenantId' },
  children: [
    {
      label: 'External Docs',
      pathSegment: 'docs',
      externalLink: {
        url: 'https://example.com/{context.tenant}/docs'
      }
    }
  ]
}
```

1. Navigate to `/home/tenantA/overview`
2. Click "External Docs" → opens `https://example.com/tenantA/docs` ✅
3. Navigate to `/home/tenantB/overview`
4. Click "External Docs" → opens `https://example.com/tenantA/docs` ❌ (expected `tenantB`)

#### Root cause

`node.externalLink` is a reference to the cached navigation tree. Assigning `.url` overwrites the template in-place. The old core avoids this because `getRouteLink()` processes the URL into a local variable without touching the original node.

**Comment by JohannesDoberer** [comment]
Still having the issue with context
```
{
              pathSegment: 'home',
              label: 'Home',
              viewUrl: 'http://localhost:4400/microfrontend.html#home',
              children: [
                {
                  pathSegment: ':tenantId',
                  context: { tenant: ':tenantId' },
                  children: [
                    {
                      pathSegment: 'overview',
                      label: 'Overview',
                      viewUrl: 'http://localhost:4400/microfrontend.html#overview',
                    },
                    {
                      label: 'External Docs',
                      pathSegment: 'docs',
                      externalLink: {
                        url: 'https://example.com/{context.tenant}/docs'
                      }
                    }
                  ]
                }
              ]
            },
```
~~https://example.com/%7Bcontext.tenant%7D/docs~~ -> probably cache issue from my side
...but if `addNavHrefs:true` is set the link is still wrong

Bildschirmfoto 2026-07-29 um 17 03 42

File: core-modular/test/services/navigation.service.spec.ts [hunk 1]
```diff
@@ -503,6 +503,65 @@ describe('NavigationService', () => {
...
+
+      expect(openExternalLinkSpy).toHaveBeenCalledWith(
+        {
+          url: 'https://sap.com/en?foo=bar'
+        },
+        {}
+      );
+      expect(navigateSpy).not.toHaveBeenCalled();
+    });
+
```
Comment (JohannesDoberer) [inline core-modular/test/services/navigation.service.spec.ts:564]: Consider adding tests for two additional edge cases:

1. **`:pathParam` in external link URL** — e.g. `url: 'https://example.com/:tenantId/docs'` with `pathParams: { tenantId: 'myTenant' }` to verify that colon-prefixed path parameters are substituted correctly.

2. **Context merge via `pathData.selectedNode.context`** — test `navItemClick` with a `pathData` that has `selectedNode.context` set (simulating a parent node's accumulated context being merged into the external link node). This covers the `if (pathData?.selectedNode?.context)` branch at line 528.

**Review by JohannesDoberer — COMMENTED** [review]
Besides missing tests, looks good for me

**Review by JohannesDoberer — APPROVED** [review]
LGTM 👍
